python/lsst/ts/wep/task/calcZernikesTask.py

```diff
@@ -327,45 +331,66 @@ def createZkTable(
                     row[f"{foc}_{key.lower()}"] = val
             zkTable.add_row(row)
 
-        zkTable.meta["intra"] = {}
-        zkTable.meta["extra"] = {}
+        zkTable.meta = self.createZkTableMetadata()
+
+        return zkTable
+
+    def createZkTableMetadata(self):
+        """Create the metadata for the Zernike table.
+
+        Returns
+        -------
+        metadata : dict
+            Metadata for the Zernike table
+        """
+        meta = {}
+        meta["intra"] = {}
+        meta["extra"] = {}
+        cam_name = None
+
+        if self.stampsIntra is None and self.stampsExtra is None:
+            raise ValueError(
+                "No stamps available. Cannot create metadata."
+            )
 
         for dict_, stamps in [
-            (zkTable.meta["intra"], intraStamps),
-            (zkTable.meta["extra"], extraStamps),
+            (meta["intra"], self.stampsIntra),
+            (meta["extra"], self.stampsExtra),
         ]:
-            dict_["det_name"] = "" if len(stamps) == 0 else stamps.metadata["DET_NAME"]
-            dict_["visit"] = "" if len(stamps) == 0 else stamps.metadata["VISIT"]
-            dict_["dfc_dist"] = "" if len(stamps) == 0 else stamps.metadata["DFC_DIST"]
-            dict_["band"] = "" if len(stamps) == 0 else stamps.metadata["BANDPASS"]
+            if stamps is None:
```

Comment:
As this seems to imply that there won't be a `.metadata['DET_NAME']` for empty stamps...

Reply:
Should only come into play when we are using unpaired task.